### PR TITLE
Worldgen

### PR DESCRIPTION
**Pull request by PatrikLundell**
Updated wordgen with the info from issues #163 and #207

### CODE REVIEW

File: df.world.xml [hunk 1]
```diff
@@ -948,13 +948,13 @@
...
-            <bool name='made_cave_pops'/>
-            <bool name='made_cave_civs'/>
-            <bool name='placed_civs'/>
+            <bool name='place_caves'/>
+            <bool name='place_good_evil'/>
+            <bool name='place_megabeasts'/>
+            <bool name='place_other_beasts'/>
+            <bool name='make_cave_pops'/>
+            <bool name='make_cave_civs'/>
+            <bool name='place_civs'/>
```
Comment (lethosor) [inline df.world.xml]: The former names seem appropriate here - that is, some of these flags are false until the corresponding steps in worldgen run.

File: df.world.xml [hunk 2]
```diff
@@ -1120,7 +1120,7 @@
                 <int32_t name='disturbance_interaction_number'/>
                 <int32_t name='evil_cloud_number'/>
                 <int32_t name='evil_rain_number'/>
-                <int32_t since='v0.40.01'/>
+                <int32_t name='generate_divine_materials'/>
```
Comment (lethosor) [inline df.world.xml]: Leaving the `since` attribute would be better (it sounds like it's accurate from the name).

**Comment by PatrikLundell** [comment]
Regarding the worldgen names: My point is that the flags toggle to true not when the step has been performed, but at the time the step begins: Looking for "civilized" civs when "place_civs" turns to true is futile, as the first one is just about to be placed. I'm open for alternative names that better reflect the status indicated for the flags, though.

Regarding the since attribute: If you mean what I think, namely that the field should be named and the since attribute should be kept I agree. It's better to keep that meta info.

File: df.world.xml (hunk 1 above)
Comment (lethosor) [reply df.world.xml]: Looks like nothing actually uses these fields, so fine with me. Your observations appear to be correct for `place_caves` and `place_megabeasts`, at least.

**Comment by lethosor** [comment]
@PatrikLundell another thing - if you can include [these keywords](https://help.github.com/articles/closing-issues-using-keywords/) in commit messages for issues that you're fixing, they'll get closed automatically, which would be convenient.

**Pull request merged**